src/cmd/scan.go

```diff
@@ -4,50 +4,532 @@
+			if err != nil {
+				return err
+			}
+		}
+		_, err := scanZarfYamlImages(outputDirectory)
+		return err
+	},
+}
 
 var compareCmd = &cobra.Command{
 	Use:   "compare-scans BASE_SCAN NEW_SCAN",
 	Short: "Compare two grype scans using the cyclonedx-json output format",
 	Args:  cobra.ExactArgs(2),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		baseScan, newScan, err := compare.LoadScans(args[0], args[1])
+		baseScanPath := args[0]
+		newScanPath := args[1]
+		markdownTable, err := compareScans(baseScanPath, newScanPath)
 		if err != nil {
 			return err
 		}
 
-		baseScan.Metadata.Component.Name = compare.TrimDockerRegistryPrefixes(baseScan.Metadata.Component.Name)
-		newScan.Metadata.Component.Name = compare.TrimDockerRegistryPrefixes(newScan.Metadata.Component.Name)
+		fmt.Println(markdownTable)
+		return nil
+	},
+}
 
-		if baseScan.Metadata.Component.Name != newScan.Metadata.Component.Name {
-			if !allowDifferentImages {
-				return fmt.Errorf("these scans are not for the same image: %s != %s", baseScan.Metadata.Component.Name, newScan.Metadata.Component.Name)
-			} else {
-				fmt.Fprintf(os.Stderr, "Warning: these scans are not for the same image: %s != %s\n", baseScan.Metadata.Component.Name, newScan.Metadata.Component.Name)
-			}
+func compareScans(baseScanPath string, newScanPath string) (string, error) {
+	baseScan, newScan, err := compare.LoadScans(baseScanPath, newScanPath)
+	if err != nil {
+		return "", err
+	}
+
+	baseScan.Metadata.Component.Name = compare.TrimDockerRegistryPrefixes(baseScan.Metadata.Component.Name)
+	newScan.Metadata.Component.Name = compare.TrimDockerRegistryPrefixes(newScan.Metadata.Component.Name)
+
+	if baseScan.Metadata.Component.Name != newScan.Metadata.Component.Name {
+		if !allowDifferentImages {
+			return "", fmt.Errorf("these scans are not for the same image: %s != %s", baseScan.Metadata.Component.Name, newScan.Metadata.Component.Name)
+		} else {
+			_, _ = fmt.Fprintf(os.Stderr, "Warning: these scans are not for the same image: %s != %s\n", baseScan.Metadata.Component.Name, newScan.Metadata.Component.Name)
 		}
+	}
+
+	vulnStatus := compare.GenerateComparisonMap(baseScan, newScan)
+
+	markdownTable, err := compare.GenerateComparisonMarkdown(baseScan, newScan, vulnStatus)
+	if err != nil {
+		return "", err
+	}
+	return markdownTable, nil
+}
 
-		vulnStatus := compare.GenerateComparisonMap(baseScan, newScan)
+var scanAndCompareCmd = &cobra.Command{
+	Use: "scan-and-compare",
+	Short: "Scan the current Zarf package, scan the last released package, and compare the scans." +
+		"This command is a combination of `scan`, `scan-released`, and `compare-scans`.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if outputDirectory == "" {
+			var err error
+			outputDirectory, err = os.MkdirTemp("", "scans")
+			logger.Info("Output directory", slog.String("dir", outputDirectory))
+			if err != nil {
+				return err
+			}
```

Comment:
Inverse the order, first check error then log the information. What's the purpose in doing this in the above order?